### PR TITLE
Added the ctrl_state parameter to mcx() function

### LINKED ISSUE #12044: Addition of ctrl_state Parameter to mcx() Function
### Feature Description

Currently, the mcx function in the quantum circuit library lacks a ctrl_state parameter, which limits its flexibility in certain use cases. This feature request proposes the addition of a ctrl_state parameter to the mcx function, similar to what is present in the ch and cy functions.

### Proposed Change:
Modify the mcx function to include a ctrl_state parameter, allowing users to specify the control state of the multi-controlled X gate. This parameter should accept either a decimal value or a bitstring (e.g., '1') and default to controlling the '1' state if not provided.

### Expected Behavior:
Upon calling the mcx function, users can pass the ctrl_state parameter to specify the control state for the multi-controlled X gate. This parameter will enhance the functionality and flexibility of the function, enabling users to perform more complex quantum operations.

### Implementation Details:
The mcx function should be modified to accept the ctrl_state parameter. This parameter should then be passed to the appropriate gate implementation for constructing the multi-controlled X gate.

### PR DESCRIPTION
**Pull request by elucidator8918**
### Description:

This PR addresses a feature request (#12044) to enhance the functionality of the mcx function in the quantum circuit library. Currently, the mcx function lacks a ctrl_state parameter, limiting its flexibility in certain quantum operations. 

The proposed change modifies the mcx function to include a ctrl_state parameter, allowing users to specify the control state of the multi-controlled X gate. This parameter can accept either a decimal value or a bitstring and defaults to controlling the '1' state if not provided.

### Expected Behavior:
Upon calling the mcx function, users can now pass the ctrl_state parameter to specify the control state for the multi-controlled X gate. This enhancement significantly enhances the functionality and flexibility of the function, enabling users to perform more complex quantum operations efficiently.

### Implementation Details:
- Modified the mcx function to accept the ctrl_state parameter.

@ShellyGarion: Please review the changes and provide feedback. Thank you :D

### CODE REVIEW

**Comment by ShellyGarion** [comment]
@elucidator8918 - Thanks for your contribution to Qiskit. 
It seems that you have some lint (black) errors.
Could you add some tests [here](https://github.com/Qiskit/qiskit/blob/main/test/python/circuit/test_controlled_gate.py) ?

**Comment by elucidator8918** [comment]
@ShellyGarion I think I know the linting mistake I made. Can I create another PR and resolve this?

**Comment by ShellyGarion** [comment]
> @ShellyGarion I think I know the linting mistake I made. Can I create another PR and resolve this?

You can add commits to this PR

**Comment by elucidator8918** [comment]
@ShellyGarion The checks got passed :) only review is required

File: qiskit/circuit/quantumcircuit.py [hunk 1]
```diff
@@ -4234,14 +4238,14 @@ def mcx(
         num_ctrl_qubits = len(control_qubits)
 
         available_implementations = {
-            "noancilla": MCXGrayCode(num_ctrl_qubits),
-            "recursion": MCXRecursive(num_ctrl_qubits),
-            "v-chain": MCXVChain(num_ctrl_qubits, False),
-            "v-chain-dirty": MCXVChain(num_ctrl_qubits, dirty_ancillas=True),
+            "noancilla": MCXGrayCode(num_ctrl_qubits,ctrl_state=ctrl_state),
```
Comment (ShellyGarion) [inline qiskit/circuit/quantumcircuit.py]: I think there should be a space after the comma (CI checks seem to be stuck now from some reason).
This is relevant for the lines below as well.

**Review by ShellyGarion — CHANGES_REQUESTED** [review]
Could you add some tests [here](https://github.com/Qiskit/qiskit/blob/main/test/python/circuit/test_controlled_gate.py) ?

**Comment by elucidator8918** [comment]
@ShellyGarion added some unit tests 

waiting for review :)

**Review by ShellyGarion — CHANGES_REQUESTED** [review]
Please also add release notes for the new feature.

File: test/python/circuit/test_controlled_gate.py [hunk 2]
```diff
@@ -970,6 +970,25 @@ def test_ccx_ctrl_state_consistency(self):
         ref_circuit.append(ccx, [qreg[0], qreg[1], qreg[2]])
         self.assertEqual(qc, ref_circuit)
 
+    @data((4, [0, 1, 2], 3, "010"), (4, [2, 1, 3], 0, "010"))
+    @unpack
+    def test_multi_control_x_ctrl_state_parameter(
+        self, num_qubits, ctrl_qubits, target_qubit, ctrl_state
+    ):
+        """To check the consistency of parameters ctrl_state in MCX
+        See issue: https://github.com/Qiskit/qiskit-terra/issues/12050
```
Comment (ShellyGarion) [inline test/python/circuit/test_controlled_gate.py]: Note that it's a PR and not an issue. We usually don't add the links.

File: test/python/circuit/test_controlled_gate.py [hunk 3]
```diff
@@ -970,6 +970,25 @@ def test_ccx_ctrl_state_consistency(self):
         ref_circuit.append(ccx, [qreg[0], qreg[1], qreg[2]])
         self.assertEqual(qc, ref_circuit)
 
+    @data((4, [0, 1, 2], 3, "010"), (4, [2, 1, 3], 0, "010"))
```
Comment (ShellyGarion) [inline test/python/circuit/test_controlled_gate.py]: perhaps one of the "010" could be an int and not a string?

File: releasenotes/notes/added-parameter-ctrl_state-mcx-816dcd80e459a5ed.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+features:
+  - |
+    Added ctrl_state parameter to mcx() function.
```
Comment (ShellyGarion) [inline releasenotes/notes/added-parameter-ctrl_state-mcx-816dcd80e459a5ed.yaml]: mcx() should be written as :func:`.QuantumCircuit.mcx()` 
also ctrl_state should be emphasized accordingly
(also in the other places in the release notes)

**Review by ShellyGarion — CHANGES_REQUESTED** [review]
LGTM, I only have a few minor comments.

**Review by ShellyGarion — APPROVED** [review]
LGTM! thanks @elucidator8918

**Review by alexanderivrii — APPROVED** [review]
Thank you for the contribution! Given Shelly's approval, this can be merged,

**Pull request merged**